### PR TITLE
Updated notebook to run on TensorFlow 2.18.0

### PR DESCRIPTION
**Pull request by seyedrezamirkhani**
**Updated regression.ipynb**

-  When calling the fit method, TensorFlow complains that the data isn't in the correct input shape. The solution is to put the numeric input parameter, which is an array of numbers, as the last input parameter for the model.

- The y_eval and y_train parameters to the fit method need to be 2 dimensional arrays when used with the ziln loss function. I've updated this code block to reflect this. 

- Updated the _script_ block to a _bash_ block, as _script_ it is no longer supported. In doing this, the code around downloading and unzipping the data files has been modified to only run if the files do not exist already.

- Reused DATA_FOLDER variable where possible to avoid hard coding paths.

- Changed the default path for the related data and output files to be in the local tmp directory where the notebook is stored. This avoids problems with people not having access to the default root tmp folder. The path name has also been made shorter.

**Update to setup.py**

update package name to scikit-learn

**Added .gitignore** 

This file includesjupyternotebook/python and tmp folder created by regression.ipynb

Thank you for writing your excellent paper+code and sharing it along with the data!

### CODE REVIEW

**Comment by miaojingang** [comment]
Thanks @seyedrezamirkhani for the PR. Unfortunately the original owners all have either left Google or moved to different teams and do not have the privilege needed to approve PRs. I'd develop in a fork instead.

**Comment by seyedrezamirkhani** [comment]
Thanks for your response, it is a great motivator. I've updated my fork now and also put a link to it in papers with code. You guys did an awsome job in devising a method and explaining it well together with code + data. It's a home run as far as I'm concerned and the praise is well deserved!

By the way, I live and grew up in the UK and used your method in my previous job for a customer in the gaming industry. My linked in profile is https://www.linkedin.com/in/reza-mirkhani-3270951/ and I'm happy to connect.
Kind regards
Reza

    On Thursday 5 December 2024 at 01:45:06 GMT, Jingang Miao ***@***.***> wrote:  
 
 


Thanks @seyedrezamirkhani for the PR. Unfortunately the original owners all have either left Google or moved to different teams and do not have the privilege needed to approve PRs. I'd develop in a fork instead.
